### PR TITLE
:bug: stop rendering manifests when addon configured condition is not true

### PR DESCRIPTION
**Pull request by haoqing0110**
## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/addonmanager/controllers/agentdeploy/controller.go [hunk 1]
```diff
@@ -381,6 +381,12 @@ func (c *addonDeployController) buildDeployManifestWorksFunc(addonWorkBuilder *a
 			return nil, nil, fmt.Errorf("failed to get agentAddon")
 		}
 
+		if agentAddon.GetAgentAddonOptions().ConfigCheckEnabled &&
+			!meta.IsStatusConditionTrue(addon.Status.Conditions, addonapiv1alpha1.ManagedClusterAddOnConditionConfigured) {
+			klog.InfoS("Addon configured condition is not set in status", "addonName", addon.Name)
+			return nil, nil, fmt.Errorf("addon %s configured condition is not is not set in status", addon.Name)
```
Comment (qiujian16) [inline pkg/addonmanager/controllers/agentdeploy/controller.go]: do we need to return error here? I think we do not, the update of addon will retrigger the reconcile.

File: pkg/addonmanager/controllers/agentdeploy/controller.go [hunk 2]
```diff
@@ -443,6 +449,12 @@ func (c *addonDeployController) buildHookManifestWorkFunc(addonWorkBuilder *addo
 			return nil, fmt.Errorf("failed to get agentAddon")
 		}
 
+		if agentAddon.GetAgentAddonOptions().ConfigCheckEnabled &&
+			!meta.IsStatusConditionTrue(addon.Status.Conditions, addonapiv1alpha1.ManagedClusterAddOnConditionConfigured) {
+			klog.InfoS("Addon configured condition is not set in status", "addonName", addon.Name)
+			return nil, fmt.Errorf("addon %s configured condition is not is not set in status", addon.Name)
```
Comment (qiujian16) [inline pkg/addonmanager/controllers/agentdeploy/controller.go]: ditto

File: pkg/addonmanager/controllers/agentdeploy/controller.go (hunk 1 above)
Comment (haoqing0110) [reply pkg/addonmanager/controllers/agentdeploy/controller.go]: fixed.

File: pkg/addonmanager/controllers/agentdeploy/controller.go (hunk 2 above)
Comment (haoqing0110) [reply pkg/addonmanager/controllers/agentdeploy/controller.go]: fixed.

**Pull request merged**